lib/core/xskdev/xskdev.c

```diff
@@ -988,12 +993,20 @@ xskdev_socket_destroy(xskdev_info_t *xi)
         CNE_DEBUG("ifindex %d, %s, prog_id %u\n", xi->if_index, xi->ifname, xi->prog_id);
         if (xi->if_index) {
             if (xi->unprivileged == 0) {
+#ifdef USE_LIBBPF_8
```

Comment:
The set10() sets the define to true or false, which means the defines are always created in builddir/cne_build_config.h and we have to use #if USE_XYZ instead of #ifdef USE_XYZ as this will always be true.

Reply:
Yeah, I could see the defines in the builddir/cne_build_config.h and originally(when I was using the set10s) the code here was using  #if USE_XYZ but these were throwing up the error below... let me play around with it a little more and see if I missed something else

```
FAILED: lib/core/xskdev/libcne_xskdev.so.p/xskdev.c.o
cc -Ilib/core/xskdev/libcne_xskdev.so.p -Ilib/core/xskdev -I../lib/core/xskdev -Ilib/include -I../lib/include -Ilib/core/osal -I../lib/core/osal -Ilib/core/log -I../lib/core/log -I. -I.. -Ilib/core/cne -I../lib/core/cne -Ilib/common/uds -I../lib/common/uds -Ilib/core/mmap -I../lib/core/mmap -Ilib/core/mempool -I../lib/core/mempool -Ilib/core/pktmbuf -I../lib/core/pktmbuf -I/usr/include/libnl3 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -Werror -O3 -g -muintr -Wno-pedantic -Wcast-qual -Wdeprecated -Wformat-nonliteral -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpointer-arith -Wsign-compare -Wstrict-prototypes -Wundef -Wwrite-strings -Wno-address-of-packed-member -Wno-packed-not-aligned -Wno-missing-field-initializers -D_GNU_SOURCE -include cne_build_config.h -march=native -DCC_AVX2_SUPPORT -fPIC -MD -MQ lib/core/xskdev/libcne_xskdev.so.p/xskdev.c.o -MF lib/core/xskdev/libcne_xskdev.so.p/xskdev.c.o.d -o lib/core/xskdev/libcne_xskdev.so.p/xskdev.c.o -c ../lib/core/xskdev/xskdev.c
../lib/core/xskdev/xskdev.c: In function 'xskdev_socket_create':
../lib/core/xskdev/xskdev.c:959:5: error: "USE_LIBBPF_8" is not defined, evaluates to 0 [-Werror=undef]
  959 | #if USE_LIBBPF_8
      |     ^~~~~~~~~~~~
../lib/core/xskdev/xskdev.c: In function 'xskdev_socket_destroy':
../lib/core/xskdev/xskdev.c:996:5: error: "USE_LIBBPF_8" is not defined, evaluates to 0 [-Werror=undef]
  996 | #if USE_LIBBPF_8
      |     ^~~~~~~~~~~~
../lib/core/xskdev/xskdev.c:1005:5: error: "USE_LIBBPF_8" is not defined, evaluates to 0 [-Werror=undef]
 1005 | #if USE_LIBBPF_8
      |     ^~~~~~~~~~~~
cc1: all warnings being treated as errors
```

Reply:
Is cne_build_config.h not getting included ?

The only other thing I can think of is verify spelling, as it normally messes me up.

Reply:
it was a homer (simpson) moment on my part - I had a mix of #ifdefs and #ifs but I've cleaned it all up now. so I'm using the set10 function and #ifs throughout

Reply:
Thanks for all the pointers @KeithWiles